src/platform/input/game-pads.js

```diff
@@ -68,62 +109,636 @@ const MAPS = {
+         * @type {object}
+         */
+        this.map = map;
+
+        /**
+         * The hand this gamepad is usually handled on. Only relevant for XR pads. Value is either "left", "right" or "none".
+         *
+         * @type {string}
+         */
+        this.hand = gamepad.hand || 'none';
+
+        /**
+         * The original Gamepad API gamepad.
+         *
+         * @type {Gamepad}
+         * @ignore
+         */
+        this.pad = gamepad;
+    }
+
+    /**
+     * Whether the gamepad is connected
+     *
+     * @type {boolean}
+     */
+    get connected() {
+        return this.pad.connected;
+    }
+
+    /**
+     * Update the existing GamePad Instance.
+     *
+     * @param {Gamepad} gamepad - The original Gamepad API gamepad.
+     * @ignore
+     */
+    update(gamepad) {
+        this.pad = gamepad;
+
+        const buttons = this._buttons;
+        for (let i = 0, l = buttons.length; i < l; i++) {
+            const button = buttons[i];
+
+            if (button)
+                button.update(gamepad.buttons[i]);
+        }
+
+        return this;
+    }
+
+    /**
+     * Update the previous values for buttons and axes.
+     *
+     * @ignore
+     */
+    updatePrevious() {
+        const buttons = this._buttons;
+        for (let i = 0, m = buttons.length; i < m; i++) {
+            buttons[i].updatePrevious();
+        }
+
+        // Store previous values for axes for dual buttons.
+        const previousAxes = this._previousAxes;
+        previousAxes.length = 0;
+        previousAxes.push(...this.pad.axes);
+    }
+
+    /**
+     * The values from analog axes present on the GamePad. Values are between -1 and 1.
+     *
+     * @type {number[]}
+     */
+    get axes() {
+        return this.map.axes ? this.map.axes.map(a => this.pad.axes[MAPS_INDEXES.axes[a]] || 0) : [];
+    }
+
+    /**
+     * The buttons present on the GamePad. Some buttons may be null.
+     *
+     * @type {GamePadButton[]}
+     */
+    get buttons() {
+        return this.map.buttons ? this.map.buttons.map(b => this._getButton(b)) : [];
+    }
+
+    /**
+     * Make the gamepad vibrate.
+     *
+     * @param {number} intensity - Intensity for the vibration in the range 0 to 1.
+     * @param {number} duration - Duration for the vibration in milliseconds.
+     * @param {object} [options] - Options for special vibration pattern.
+     * @param {string} [options.type] - Type of pattern. Available types are "dual-rumble" and "vibration". Defaults to "vibration".
+     * @param {number} [options.startDelay] - Delay before the pattern starts, in milliseconds. Defaults to 0.
+     * @param {number} [options.strongMagnitude] - Intensity for strong actuators in the range 0 to 1. Defaults to intensity.
+     * @param {number} [options.weakMagnitude] - Intensity for weak actuators in the range 0 to 1. Defaults to intensity.
+     * @returns {Promise<boolean>} Return a Promise resulting in true if the pulse was successfully completed.
```

Comment:
`actuator.playEffect` on Chrome returns a string:

![image](https://user-images.githubusercontent.com/5236548/219868782-63c0d7b4-d1aa-4438-a3c3-bbd89969c7f6.png) 

(`"preempted"` happens if you plug the USB out while it vibrates)

I can't test Firefox, because it doesn't even give access to `vibrationActuator`.

I don't know when this was changed, at least this link says it should be a boolean, yet it isn't: https://developer.mozilla.org/en-US/docs/Web/API/GamepadHapticActuator/playEffect

Reply:
Nice spot! I fixed it by forcing into boolean and added support for Mozilla.